server/cluster/coordinator.go

```diff
@@ -966,3 +967,17 @@ func (c *coordinator) getPausedSchedulerDelayUntil(name string) (int64, error) {
 	}
 	return s.GetDelayUntil(), nil
 }
+
+func (c *coordinator) CheckTransferWitnessLeader(region *core.RegionInfo) {
+	c.RLock()
+	defer c.RUnlock()
+	if s, ok := c.schedulers[schedulers.TransferWitnessLeaderName]; ok {
+		if schedulers.NeedTransferWitnessLeader(region) {
```

Comment:
How about checking it first before locking the coordinator?

Reply:
Sure, fixed, thanks!